### PR TITLE
Documentation of click based commands from docstrings

### PR DESCRIPTION
**Pull request by PrajwalPrashanth**
- [x] Add sphinx click extension
- [x] Review and modify docstring contents(add helper texts)
- [x] Style Chnages
- [x] Test cli commands and modify examples

### CODE REVIEW

**Comment by PrajwalPrashanth** [comment]
@sidujjain Check this out its ready, local build is able to generate the page but there is some issue with ReadTheDocs and sphinx-click.

**Comment by PrajwalPrashanth** [comment]
@sidujjain changes are done. Here is the preview link
https://prajwalprashanth-demo.readthedocs.io/en/pp-ch1321-cli-docs/api-reference/cli.html

File: docs/api-reference/cli.md [hunk 1]
```diff
@@ -1,72 +1,10 @@
...
-By default, the jovian jupyter extension is enabled.
-
-```
-$ jovian enable-extension
-```
-
-```
-$ jovian disable-extension
-```
+## CLI Commands
```
Comment (aakashns) [inline docs/api-reference/cli.md:1]: Should add some general explanation about what the CLI does, how to install it etc. where to use it (terminal on Mac/Linux & Command Prompt/Conda Prompt on Windows) etc.

File: requirements.txt [hunk 2]
```diff
@@ -45,29 +54,35 @@ pkginfo==1.5.0.1
...
 requests==2.22.0
 requests-toolbelt==0.9.1
 restructuredtext-lint==1.3.0
 rope==0.14.0
+Send2Trash==1.5.0
 six==1.12.0
 snowballstemmer==1.9.1
 Sphinx==2.2.0
 sphinx-autobuild==0.7.1
+git+https://github.com/PrajwalPrashanth/sphinx-click.git
```
Comment (sidujjain) [inline requirements.txt:85]: let's move the fork under JovianML

File: jovian/cli.py [hunk 3]
```diff
@@ -138,12 +137,12 @@ def exec_pull(ctx, notebook, version):
...
 @click.pass_context
 def set_project(ctx,  notebook, project):
-    """Associate notebook (filename.ipynb) to Jovian project (username.title)
+    """Associate notebook (filename.ipynb) to Jovian project (username/title)
 
         $ jovian set-project my_notebook.ipynb danb/keras-example
 
-    This will create or update the .jovianrc file in the current directory to ensure that commits
-    inside the Jupyter notebook my_notebook.ipynb add new versions to the project danb/keras-example
+    This will create or update the .jovianrc file in the current directory to ensure that commit
```
Comment (sidujjain) [inline jovian/cli.py:144]: this could be reverted. you can add quotes around my_notebook.ipynb

File: jovian/cli.py [hunk 4]
```diff
@@ -109,23 +110,21 @@ def exec_clone(ctx, notebook, version, no_outputs, overwrite):
...
+        $ jovian clone -v 10 aakashns/jovian-tutorial 
     """
-
     clone(slug=notebook, version=version, include_outputs=not no_outputs, overwrite=overwrite)
 
 
 @main.command("pull", short_help="Fetch new version of notebook hosted Jovian.")
-@click.option('-n', '--notebook', 'notebook')
-@click.option('-v', '--version', 'version')
+@click.option('-n', '--notebook', 'notebook', help="Notebook project (format: username/notebook-title)")
```
Comment (sidujjain) [inline jovian/cli.py:119]: nit: username/title

File: jovian/cli.py [hunk 5]
```diff
@@ -97,10 +98,10 @@ def activate_env(ctx):
 
 
 @main.command("clone", short_help="Clone a notebook hosted on Jovian")
-@click.argument('notebook')
-@click.option('-v', '--version', 'version')
-@click.option('--no-outputs', 'no_outputs', is_flag=True, default=False)
-@click.option('--overwrite', 'overwrite', is_flag=True)
+@click.argument('notebook', metavar="<username/notebook-title>")
```
Comment (sidujjain) [inline jovian/cli.py:101]: let's keep `username/title` for consistency

File: jovian/cli.py [hunk 6]
```diff
@@ -109,23 +110,21 @@ def exec_clone(ctx, notebook, version, no_outputs, overwrite):
 
     Or clone a specific version of notebook:
 
-        $ jovian clone aakashns/jovian-tutorial -v 10
```
Comment (sidujjain) [inline jovian/cli.py:112]: this can be reverted

File: jovian/cli.py [hunk 7]
```diff
@@ -47,7 +47,7 @@ def help(ctx):  # no-cover
 @main.command('version')
 @click.pass_context
 def main_version(ctx):  # no-cover
-    """Print Jovian’s version number."""
+    """Print installed jovian library version."""
```
Comment (sidujjain) [inline jovian/cli.py:50]: Jovian

File: jovian/cli.py (hunk 6 above)
Comment (PrajwalPrashanth) [reply jovian/cli.py:112]: Click's usage shows [OPTIONS] first.
image

Will it be add options after argument ?

**Comment by PrajwalPrashanth** [comment]
@sidujjain Order has been changed and [OPTIONS] is moved to last check it out
https://prajwalprashanth-demo.readthedocs.io/en/pp-ch1321-cli-docs/api-reference/cli.html

**Pull request merged**